Assets/Scripts/Entity/Player/Sanity/SanityController.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Remote;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Utils;
+using static Utils.Enums;
+
+namespace Entity.Player.Sanity
+{
+    public class SanityController : MonoBehaviour
+    {
+        public delegate void PlayerStateChanged(int newValue);
+
+        [Tooltip("100: sane, 0: insane")]
+        [Range(0, 100)]
+        [SerializeField]
+        private int sanity;
+
+        [Tooltip("Handles the math and holds stats influencing the player's sanity.")]
+        [SerializeField]
+        private SanityMath sanityMath;
+        
+        public static event PlayerStateChanged OnPlayerSanityUpdate;
+        
+        // builds up to 1 or -1 for the next tick
+        private float _nextTick;
+        
+
+        private void OnEnable()
+        {
+            RemoteStatusHandler.OnPlayerSanityRemoteUpdate += ChangePlayerSanity;
+
+            PlayerState.OnPlayerHealthUpdate += (int health) => sanityMath.InfluenceSanityByStat(StatType.Health, health);
+            PlayerState.OnPlayerSaturationUpdate += (int saturation) => sanityMath.InfluenceSanityByStat(StatType.Saturation, saturation);
+            PlayerState.OnPlayerHydrationUpdate += (int hydration) => sanityMath.InfluenceSanityByStat(StatType.Hydration, hydration);
+        }
+
+        private void OnDisable()
+        {
+            RemoteStatusHandler.OnPlayerSanityRemoteUpdate -= ChangePlayerSanity;
+
+            PlayerState.OnPlayerHealthUpdate -= (int health) => sanityMath.InfluenceSanityByStat(StatType.Health, health);
+            PlayerState.OnPlayerSaturationUpdate -= (int saturation) => sanityMath.InfluenceSanityByStat(StatType.Saturation, saturation);
+            PlayerState.OnPlayerHydrationUpdate -= (int hydration) => sanityMath.InfluenceSanityByStat(StatType.Hydration, hydration);
+        }
+
+        /// <summary>
+        /// Tick sanity.
+        /// </summary>
+        private void Update()
+        {
+            TickSanity();
+        }
+
+        private void ChangePlayerSanity(int changeBy)
+        {
+            sanity = Mathf.Clamp(sanity + changeBy, 0, 100);
+            OnPlayerSanityUpdate?.Invoke(sanity);
+        }
+
+        /// <summary>
+        /// Sums up a tick weighting the total change rate with <see cref="severity"/>. 
+        /// Only sums up a positive tick if player doesn't have full sanity. Ticks are either
+        /// positive or negative 1. If summed up tick, changes player sanity.
+        /// </summary>
+        private void TickSanity()
+        {
+            _nextTick += sanityMath.GetCurrentChange();
+            if (sanity == 100 && _nextTick > 0)
```

Comment:
Are you only checking the upper bound on purpose, or did you forget to check the lower bound when decreasing the sanity? 😅

Reply:
I check upper bound only, because I assumed the player dies when sanity goes 0. I don't know if we want that, but I'd prefer it because it really motivates you to do something when you notice mobs are spawning around you :)
Independently of whether we want that or not, I think building up a "negative malus" when sanity = 0 for the lower bound is alright as well, don't you think?

Reply:
Yeah, it's fine for now 👍